src/Raven.Server/Documents/PeriodicBackup/Restore/Sharding/ShardedRestoreOrchestrationTask.cs

```diff
@@ -64,21 +65,22 @@ protected override async Task RestoreAsync()
             _result = await RestoreOnAllShardsAsync(Progress, RestoreConfiguration, shardedDbContext);
         }
 
-        protected override void OnAfterRestore()
+        protected override async Task OnAfterRestoreAsync()
         {
             DisableOngoingTasksIfNeeded(RestoreSettings.DatabaseRecord);
 
             Progress.Invoke(Result.Progress);
 
             // after the db for restore is done, we can safely set the db state to normal and write the DatabaseRecord
+            DatabaseRecord databaseRecord;
             using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext serverContext))
             using (serverContext.OpenReadTransaction())
             {
-                var databaseRecord = ServerStore.Cluster.ReadDatabase(serverContext, DatabaseName);
+                databaseRecord = ServerStore.Cluster.ReadDatabase(serverContext, DatabaseName);
                 databaseRecord.DatabaseState = DatabaseStateStatus.Normal;
-
-                RestoreSettings.DatabaseRecord = databaseRecord;
             }
+
+            await SaveDatabaseRecordAsync(DatabaseName, databaseRecord, null, Result, Progress);
```

Comment:
what is `null` here?

Reply:
null is `databaseValues`
added name identifier